launcher/src/main/kotlin/com/cognifide/gradle/aem/launcher/BuildScaffolder.kt

```diff
@@ -9,9 +10,19 @@ class BuildScaffolder(private val launcher: Launcher) {
         saveProperties()
         saveSettings()
         saveRootBuildScript()
-        saveEnvBuildScript()
+        when (propertyAemVersion()){
+            "cloud" -> EnvCloudScaffolder(launcher).scaffold()
+            else -> EnvOnPremScaffolder(launcher).scaffold()
+        }
+    }
+
+    fun archetypeProperties() = Properties().apply {
+        File("archetype.properties").inputStream().buffered().use { load(it) }
     }
 
+    fun propertyAemVersion() = archetypeProperties().getProperty("aemVersion")
+        ?: error("Cannot read aemVersion from file 'archetype.properties'!")
```

Comment:
please always provide a message that is contextual and descriptive for potential user.

btw you should not throw error but .. when that property is missing then it means that no env files should be generated.